### DISCUSSION TITLE
Message.lvclass an interface

### DISCUSSION THREAD

**Discussion by CaseyZS** [ni/actor-framework#60] [General]
**Is your feature request related to a problem? Please describe.**
Not a problem really, but there's no reason that Message.lvclass can't be an interface now. It could open up some neat possibilities for the future of AF.

**Describe the solution you'd like**
Make Message.lvclass an interface instead of a class. Project Providers that use Message.lvclass (e.g. Message Maker) would also need to be validated against this change.

**Describe alternatives you've considered**
Derrick (Illuminated G) already has a VIP out that does this but it could be integrated into native AF.

**Additional context**
N/A

**Comment by svelderrainruiz** [comment]
sandbox branch for @niACS  [users/asmith/message-class-interface](https://github.com/ni/actor-framework/tree/users/asmith/message-class-interface) has been created at his request, since this is going to be a long standing experiment.

**Comment by svelderrainruiz** [comment]
turning this into a discussion for now

**Comment by niACS** [comment]
We have a volunteer from the community who is willing to test this change in a live project.
